apiserver/common.go

```diff
@@ -52,10 +52,14 @@ func GetSecrets(cluster *crv1.Pgcluster, ns string) ([]msgs.ShowUserSecret, erro
 	return output, err
 }
 
-func GetPodStatus(deployName, ns string) (string, string) {
-
-	//get pods with replica-name=deployName
-	pods, err := kubeapi.GetPods(Clientset, config.LABEL_REPLICA_NAME+"="+deployName, ns)
+// GetReplicaPodStatus gets the status of all replica pods in the cluster. Specifically, using
+// the provided cluster name and namespace, it looks up all replica pod in the cluster, and then
+// provides a status for each pod ("Ready" or "Not Ready")
+func GetReplicaPodStatus(clusterName, ns string) (string, string) {
```

Comment:
Would it be too much involved of a refactor to perhaps turn this signature into something like:

```
func GetReplicaPodStatus(clusterName, ns string) (ReplicaPodStatus, error) { ... }
```

Where,

```
type ReplicaPodStatus struct {
    Name string
    Ready bool
}
```

So, that:

```
if c.Name == "database" {
    return ReplicaPodStatus{Name: nodeName, Status: c.Ready}, nil
}
```

And:

```
var (
    ErrReplicaNotFound = errors.New("replica not found")
    ...
)
...
func GetReplicaPodStatus(...) (ReplicaPodStatus, error) {
return ReplicPodStatus{}, ErrReplicaNotFound  // or other applicable 'error'
}
```